python/ray/logger.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import sys
+
+default_logging_level = logging.INFO
+default_logger = logging.getLogger("ray_default")
+default_logger.setLevel(default_logging_level)
+ch = logging.StreamHandler(sys.stderr)
+ch.setLevel(default_logging_level)
+default_logger.addHandler(ch)
+
+
+def create_logger(level_threshold_str,
+                  message_format=None,
+                  date_format=None,
+                  name=None):
+    if message_format is None:
+        formatter = None
+    else:
+        formatter = get_formatter(message_format, date_format)
+    return get_logger_with_formatter(level_threshold_str, formatter, name)
+
+
+def get_logger_with_formatter(level_threshold_str, formatter=None, name=None):
```

Comment:
This should probably just be `basicConfig` where you pass format string in?

I really do not think one should be configuring loggers from the module. This should be done at only one place, at the entry point to the program. Here, this function, not just creates a logger but also attaches handlers and configures logging.

Now, anyone using Ray as a module, good luck first going over all Ray loggers and cleanup this first.

Reply:
1. If only the entry point can do the configuration, how can you use different logging levels in different modules as you said in previous comment?
2. Users may want to use his own format, do they need to pass the formatter to all modules?